### PR TITLE
docs(transformer): add documentation for `logical-assignment-operators` plugin

### PR DESCRIPTION
**Pull request by Dunqing**
follow-up #4881

### CODE REVIEW

**Comment by Dunqing** [comment]
* **#5014** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5014?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5013** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5013?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5012** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5012?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/oxc_transformer/src/es2021/logical_assignment_operators.rs [hunk 1]
```diff
@@ -1,3 +1,59 @@
...
+//! ```js
+//! a ??= b;
+//! obj.a.b ??= c;
+//! ```
+//!
+//! Output:
+//! ```js
+//! var _obj$a, _obj$a2;
+//! var _a, _obj$a, _obj$a$b;
+
```
Comment (overlookmotel) [inline crates/oxc_transformer/src/es2021/logical_assignment_operators.rs:43]: Line break shouldn't be here.

File: crates/oxc_transformer/src/es2021/logical_assignment_operators.rs [hunk 2]
```diff
@@ -1,3 +1,59 @@
...
+//! Input:
+//! ```js
+//! a ??= b;
+//! obj.a.b ??= c;
+//! ```
+//!
+//! Output:
+//! ```js
+//! var _obj$a, _obj$a2;
+//! var _a, _obj$a, _obj$a$b;
```
Comment (overlookmotel) [inline crates/oxc_transformer/src/es2021/logical_assignment_operators.rs:42]: I think only the 2nd line should be here.
```suggestion
//! var _a, _obj$a, _obj$a$b;
```

File: crates/oxc_transformer/src/es2021/logical_assignment_operators.rs (hunk 1 above)
Comment (Dunqing) [reply crates/oxc_transformer/src/es2021/logical_assignment_operators.rs:42]:
```suggestion
//!
```

Comment (Dunqing) [reply crates/oxc_transformer/src/es2021/logical_assignment_operators.rs]: Thanks!

**Review by overlookmotel — APPROVED** [review]

**Pull request merged**